### DISCUSSION TITLE
App Builder - TileServices-SideBySide

### DISCUSSION THREAD

**Discussion by Andreas-Lampert** [e-netze/webgis-community#196] [Q&A]
Hallo Jürgen!

Ich würde gerne mittels App Builder - TileServices-SideBySide vergleiche der diversen Orthofoto-Epochen bereitstellen. 
Die Konfiguration wurde im WebGIS5 vorgenommen und funktioniert mit Einschränkungen.

Beim ersten Aufruf der App aus dem Portal ist kein Bild sichtbar. Es scheint ganz oben links ein Icon zu sein, welches auf ein nicht vorhandenes Bild hindeutet (verdeckt durch das weiße Kästchen).
![image](https://github.com/user-attachments/assets/855a5f13-a64d-4398-a778-ceba68a4bee5)

Sobald ich 1x auf Zoom in oder Zoom out klicke wird das Bild geladen und ist problemlos verwendbar.
![image](https://github.com/user-attachments/assets/413d4761-77b4-4855-88a6-daaecf571965)

Ist beim ersten Aufruf der Karte irgendwas in der Konfig zu berücksichtigen, damit die Services sofort geladen werden?
Gibt es bei den TileServices-SideBySide auch die Möglichkeit links und rechts eine Bezeichnung zu vergeben (z.B. Orthofoto aktuell - Orthofoto 2005). Das sind im Moment zwei leere weiße Kästchen und ich finde keine Möglichkeit, diese im App Builder zu konfigurieren. 

Im WebGIS6 funktioniert die App TileServices-SideBySide überhaupt nicht. Die beiden Services liegen vollständig übereinander und anstatt dem Slider wird nur die Karte verschoben (der Slider bleibt stabil in der Mitte und es wird immer nur das oben liegende Service angezeigt, das unterliegende Service wird nie sichtbar). 

Danke!

Lg, Andi

**Comment by gis-eni** [comment]
Ich gaub es gibt hier ein Problem mit der Leaflet Version. Das *SideBySide* Plugin ist schon recht alt und hat Probleme mit dem neuen Leaflet Frameworks. Bei mir funktioniert das nur mehr bis Version Leaflet 1.6.0. (Zur Info, das Plugin ist nicht von uns sondern ein Plugin von der Leaflet Seite)

WebGIS verwendet allerdings bereits neuere Versionen. 

Ein Workaround für das SideBySide Plugin ist, hier noch auf ein altes Leaflet zurück zu greifen. Ich bau das in die nächste Versionen einmal ein. Das kannst du bei dir auch ganz einfach selber probieren:

Geh ins ``_templates`` Verzeichnis im WebGIS Portal. Dort findes du Files:

* ``WebGIS-Service-SideBySide.html``
* ``WebGIS-TileServices-SideBySide.html``

Dort findest du jeweils ab Zeile 40 folgendes:

![image](https://github.com/user-attachments/assets/19de0a67-40f1-4d49-a190-74a3a1a770d8)

Die letzte Zeile musst du hier ergänzen:

``` javascript
webgis.mapFramework_version = "1.6.0";
```

Damit verwendet dieses Plugin eine alte Leaflet Version, mit der noch alles funktioniert.

**Comment by Andreas-Lampert** [comment]
Danke Jürgen! Funktioniert.

Lg, Andi
